zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -966,6 +966,117 @@ angular.module('zeppelinWebApp').controller('ParagraphCtrl', function($scope, $r
   };
 
   var setTable = function(data, refresh) {
+    function addButtonMenuEvent(button, menu) {
```

Comment:
@Peilin-Yang Right now, it seems there is a lot of functions that are not directly related to the controller in itself, and targeting mainly handsontable. (buildMenu, buildButton, isNumeric etc...)

So Having a service dedicated to Handsontable with all the utility functions we use, would make the pragraph controller way lighter.

I know there is pretty much everything mix inside that controller, but let's try to separate a bit more the logic when possible.

I'll be happy to help if you want a simple example to show how it can be done, or I can also do a PR to this one with the whole service refactoring